src/on-chain-value.ts

```diff
@@ -27,48 +26,73 @@ export const chainValueDataPoint: APIGatewayProxyHandler = async (event): Promis
     };
   }
 
-  const provider = config.providers[chainId];
-  if (!provider) {
+  const db = await initDb();
+  if(db === undefined) {
     return {
       statusCode: 500,
       headers: config.headers,
-      body: makeError("We don't have a provider for that chainId"),
+      body: makeError("An error has occurred while trying to initialize the database"),
     };
   }
 
-  const dapiServerAddress = contracts.DapiServer[chainId];
-  if (!dapiServerAddress) {
+  var currentDataFeedId = dataFeedId;
+  if(dapiName && !dataFeedId) {
+    const hashedDapiNameId = ethers.utils.formatBytes32String(dapiName);
+    
+    const operation = async () => db.query(`
+      SELECT event_data->'parsedLog'->'args'->> 1 as "dataFeedId" 
+      FROM dapi_events 
+      WHERE 
+      event_name = 'SetDapiName' AND 
+      event_data->'parsedLog'->'args'->> 0 = $1 
+      ORDER BY block DESC
+      LIMIT 1;
+    `, [hashedDapiNameId]);
+
+    const [err, firstResult] = await go(operation, { timeoutMs: 5_000, retries: 2 });
+    if (err || firstResult === undefined) {
+      return {
+        statusCode: 500,
+        headers: config.headers,
+        body: makeError("An error has occurred while querying the dataFeedId"),
+      };
+    }
+    currentDataFeedId = firstResult.rows[0].dataFeedId;
+  }
+
+  const operation = async () => db.query(`
+      SELECT event_data -> 'parsedLog' -> 'args' ->> 1 as "data"
+      FROM dapi_events 
+      WHERE 
+      chain = $1 AND
+      event_data->'parsedLog'->'args'->>0 = $2
+      ORDER by block DESC 
+      LIMIT 1;
+    `, [chainId, currentDataFeedId]
+  );
+
+  const [err, queryResult] = await go(operation, { timeoutMs: 5_000, retries: 2 });
+  if (err || queryResult === undefined) {
     return {
       statusCode: 500,
       headers: config.headers,
-      body: makeError("We don't have a dapiServer for that chainId"),
+      body: makeError("An error has occurred while querying the on-chain-value"),
     };
   }
 
-  const voidSigner = new ethers.VoidSigner(
-    ethers.constants.AddressZero,
-    new ethers.providers.JsonRpcProvider(provider)
-  );
-
-  const dapiServer = DapiServer__factory.connect(dapiServerAddress, voidSigner);
-
-  const readDataFeedId =
-    dataFeedId ??
-    ethers.utils.keccak256(ethers.utils.solidityPack(['address', 'bytes32'], [airnodeAddress, templateId]));
-
-  const operation = async () =>
-    await (dapiName ? dapiServer.readDataFeedWithDapiName(dapiName!) : dapiServer.readDataFeedWithId(readDataFeedId));
-  const [err, beaconResponse] = await go(operation, { timeoutMs: 5_000, retries: 2 });
-
-  const e = err as Error;
-  if (err) {
-    console.error(err);
-    console.error(e.stack);
+  if(queryResult.rows.length === 0) {
+    return {
+      statusCode: 500,
+      headers: config.headers,
+      body: makeError("Value not found"),
+    };
   }
 
+  const beaconResponse = JSON.parse(queryResult.rows[0].data);
```

Comment:
We expect to see this schema:
```
{"error":false,"beaconResponse":[{"type":"BigNumber","hex":"0x0f438d"},1654007048]}
```
this is an old value from the current endpoint, but we're getting this from this code:
```
{"error":false,"beaconResponse":{"hex":"0xcf14fcda525a1000","type":"BigNumber"}}
```
So it is heading in the right direction, but you're missing the timestamp. The beaconResponse should be an array of a BigNumber value and a timestamp.

You can probably modify the query to 
```
SELECT event_data -> 'parsedLog' ->> 'args'  as "data"
```
you'll have to slice out columns 1 and 2 (not 0) and you'll have to parse column 2 to a number to match the current schema, you can do that with `substring` and `parseInt` functions.